File: android/src/main/kotlin/cz/appsdevteam/gptom/utils/Logger.kt
package cz.appsdevteam.gptom

import android.util.Log

object GpTomLog {
    private var enabled: Boolean = false
    private var tag: String = "ADT_GP_TOM"

    fun configure(
        enabled: Boolean,
        tag: String = this.tag,
    ) {
        this.enabled = enabled
        this.tag = tag
    }

    fun d(msg: String, data: Any? = null,) {
        if (!enabled) return
        Log.d(tag, format(msg, data))
    }

    fun i(msg: String, data: Any? = null,) {
        if (!enabled) return
        Log.i(tag, format(msg, data))
    }

    fun w(msg: String, data: Any? = null,) {
        if (!enabled) return
        Log.w(tag, format(msg, data))
    }

    fun e(msg: String, tr: Throwable? = null,) {
        if (!enabled) return
        if (tr != null) Log.e(tag, msg, tr) else Log.e(tag, msg)
    }

    private fun format(msg: String, data: Any?): String {
        return if (data == null) msg else "$msg | data=$data"
    }
}

File: ios/Classes/utils/Logger.swift
import Foundation

final class GpTomLog {
    static var enabled = false

    static func i(_ msg: String, _ data: [String: Any?] = [:]) { log("I", msg, data) }
    static func d(_ msg: String, _ data: [String: Any?] = [:]) { log("D", msg, data) }
    static func w(_ msg: String, _ data: [String: Any?] = [:]) { log("W", msg, data) }
    static func e(_ msg: String, _ data: [String: Any?] = [:]) { log("E", msg, data) }

    private static func log(_ lvl: String, _ msg: String, _ data: [String: Any?]) {
        guard enabled else { return }
        if data.isEmpty {
            NSLog("ADT_GP_TOM [\(lvl)] \(msg)")
        } else {
            NSLog("ADT_GP_TOM [\(lvl)] \(msg) \(data)")
        }
    }
}
